From 43f07d8bda147ca3b5e2d5869836d7402ff9fef7 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 27 May 2019 18:52:15 +0200
Subject: [PATCH] ARM: irq_pipeline: fix allnoconfig build issue

---
 arch/arm/mm/fault.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mm/fault.c b/arch/arm/mm/fault.c
index a54151857ced..a261bba60e63 100644
--- a/arch/arm/mm/fault.c
+++ b/arch/arm/mm/fault.c
@@ -205,6 +205,13 @@ void show_pte(struct mm_struct *mm, unsigned long addr)
 	pr_cont("\n");
 }
 #else					/* CONFIG_MMU */
+unsigned long fault_entry(unsigned int exception, struct pt_regs *regs)
+{
+	return 0;
+}
+
+static inline void fault_exit(unsigned long combo) { }
+
 void show_pte(struct mm_struct *mm, unsigned long addr)
 { }
 #endif					/* CONFIG_MMU */
-- 
2.16.4

