From 6b0a9e7d4f9624bda7406c2d1807c395c5a26403 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 28 May 2019 17:33:45 +0200
Subject: [PATCH] ARM: dovetail: document oob_trap_notify() assumptions in
 callee

---
 arch/arm/mm/fault.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/arm/mm/fault.c b/arch/arm/mm/fault.c
index 9b4e5eab36b4..95c6e790ee01 100644
--- a/arch/arm/mm/fault.c
+++ b/arch/arm/mm/fault.c
@@ -60,11 +60,6 @@ unsigned long __fault_entry(int exception, struct pt_regs *regs)
 
 	trace_ARM_trap_entry(exception, regs);
 
-	/*
-	 * CAUTION: The co-kernel might demote the current context to
-	 * the in-band stage as a result of handling this trap,
-	 * returning with hard irqs on.
-	 */
 	oob_trap_notify(exception, regs);
 
 	flags = hard_local_irq_save();
-- 
2.16.4

