From 0df87953ccc0047a501ea5ced569b3ee79c98c2d Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 30 Jul 2019 12:58:43 +0200
Subject: [PATCH] ARM: drop useless enter_irq_pipeline()

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm/kernel/irq.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/kernel/irq.c b/arch/arm/kernel/irq.c
index f7fd27b352d3..30e83ee9d153 100644
--- a/arch/arm/kernel/irq.c
+++ b/arch/arm/kernel/irq.c
@@ -103,7 +103,6 @@ void __init init_IRQ(void)
 asmlinkage int __exception_irq_entry
 handle_arch_irq_pipelined(struct pt_regs *regs)
 {
-	enter_irq_pipeline(regs);
 	handle_arch_irq(regs);
 	return leave_irq_pipeline(regs);
 }
-- 
2.16.4

