From 180d22eda36a5aab7c77c72465f51714744439dd Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 28 May 2019 17:14:40 +0200
Subject: [PATCH] ARM: dovetail: notify co-kernel about und_usr

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm/kernel/entry-armv.S | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/kernel/entry-armv.S b/arch/arm/kernel/entry-armv.S
index 19936cb1235a..67517f26f3af 100644
--- a/arch/arm/kernel/entry-armv.S
+++ b/arch/arm/kernel/entry-armv.S
@@ -496,6 +496,18 @@ ENDPROC(__irq_usr)
 __und_usr:
 	usr_entry uaccess=0
 
+#ifdef CONFIG_DOVETAIL
+	get_thread_info tsk
+	ldr	r0, [tsk, #TI_PREEMPT]		@ get preempt count
+	tst	r0, #TI_OOB_MASK		@ oob stage?
+	beq	1f
+	mov	r0, #ARM_TRAP_UNDEFINSTR
+	mov	r1, sp				@ r1 = &regs
+	bl	__oob_trap_notify
+	uaccess_enable ip
+1:
+#endif
+
 	mov	r2, r4
 	mov	r3, r5
 
-- 
2.16.4

