From 2a8fda87db0cc75de1a8d6d5b73cd3fb0131f376 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 28 May 2019 17:33:45 +0200
Subject: [PATCH] ARM: dovetail: document oob_trap_notify() assumptions in
 callee

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm/mm/fault.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/arm/mm/fault.c b/arch/arm/mm/fault.c
index cb78b0927b45..27fc2ad3d65b 100644
--- a/arch/arm/mm/fault.c
+++ b/arch/arm/mm/fault.c
@@ -57,11 +57,6 @@ unsigned long __fault_entry(int exception, struct pt_regs *regs)
 
 	trace_ARM_trap_entry(exception, regs);
 
-	/*
-	 * CAUTION: The co-kernel might demote the current context to
-	 * the in-band stage as a result of handling this trap,
-	 * returning with hard irqs on.
-	 */
 	oob_trap_notify(exception, regs);
 
 	flags = hard_local_irq_save();
-- 
2.16.4

