From ea762b993f2b33b1e57a8162f30222ba6b48e56b Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 28 May 2019 17:33:36 +0200
Subject: [PATCH] arm64: dovetail: document oob_trap_notify() assumptions in
 callee

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm64/mm/fault.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/mm/fault.c b/arch/arm64/mm/fault.c
index c69df499c134..6128cc5b0cf0 100644
--- a/arch/arm64/mm/fault.c
+++ b/arch/arm64/mm/fault.c
@@ -90,11 +90,8 @@ unsigned long __fault_entry(unsigned int exception, struct pt_regs *regs)
 	int nosync = 1;
 
 	/*
-	 * CAUTION: The co-kernel might demote the current context to
-	 * the in-band stage as a result of handling this trap,
-	 * returning with hard irqs on. Do not propagate SEA traps to
-	 * the co-kernel in NMI mode, there is nothing it could do
-	 * about it.
+	 * Do not propagate SEA traps to the out-of-band handler in
+	 * NMI mode, there is nothing it could do about it.
 	 */
 	if (likely(exception != ARM64_TRAP_SEA || interrupts_enabled(regs)))
 		oob_trap_notify(exception, regs);
-- 
2.16.4

