From dd562a3cc1da23a643d580cbefaeb4fd17e60ba3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 30 Jul 2019 12:59:08 +0200
Subject: [PATCH] arm64: drop useless enter_irq_pipeline()

---
 arch/arm64/kernel/irq.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm64/kernel/irq.c b/arch/arm64/kernel/irq.c
index 0ec289b11693..bd15d2b1e665 100644
--- a/arch/arm64/kernel/irq.c
+++ b/arch/arm64/kernel/irq.c
@@ -43,7 +43,6 @@ int arch_show_interrupts(struct seq_file *p, int prec)
 asmlinkage int __exception_irq_entry
 handle_arch_irq_pipelined(struct pt_regs *regs)
 {
-	enter_irq_pipeline(regs);
 	handle_arch_irq(regs);
 	return leave_irq_pipeline(regs);
 }
-- 
2.16.4

