From 2dfb1f29a1bb76030ab3df8248a4b646259d5ac2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 2 Oct 2018 12:44:10 +0200
Subject: [PATCH] arm64: dovetail: enable VDSO access from the oob stage

---
 arch/arm64/kernel/vdso.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/kernel/vdso.c b/arch/arm64/kernel/vdso.c
index 2d419006ad4..662849f5bd0 100644
--- a/arch/arm64/kernel/vdso.c
+++ b/arch/arm64/kernel/vdso.c
@@ -221,6 +221,9 @@ int arch_setup_additional_pages(struct linux_binprm *bprm,
 void update_vsyscall(struct timekeeper *tk)
 {
 	u32 use_syscall = !tk->tkr_mono.clock->archdata.vdso_direct;
+	unsigned long flags;
+
+	flags = hard_cond_local_irq_save();
 
 	++vdso_data->tb_seq_count;
 	smp_wmb();
@@ -247,6 +250,8 @@ void update_vsyscall(struct timekeeper *tk)
 
 	smp_wmb();
 	++vdso_data->tb_seq_count;
+
+	hard_cond_local_irq_restore(flags);
 }
 
 void update_vsyscall_tz(void)
-- 
2.16.4

