From 557ddb253165771b762e233d2b9cb87903f36370 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 30 Jul 2019 12:59:08 +0200
Subject: [PATCH] arm64: drop useless enter_irq_pipeline()

---
 arch/arm64/kernel/irq.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm64/kernel/irq.c b/arch/arm64/kernel/irq.c
index 6f3257daa62..a0166a45251 100644
--- a/arch/arm64/kernel/irq.c
+++ b/arch/arm64/kernel/irq.c
@@ -41,7 +41,6 @@ int arch_show_interrupts(struct seq_file *p, int prec)
 asmlinkage int __exception_irq_entry
 handle_arch_irq_pipelined(struct pt_regs *regs)
 {
-	enter_irq_pipeline(regs);
 	handle_arch_irq(regs);
 	return leave_irq_pipeline(regs);
 }
-- 
2.16.4

