From 0d8252c7ca6e23ca463a6ee57d2142a77132a20c Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 20 Jan 2020 16:51:49 +0100
Subject: [PATCH] clocksource: mmio: set vDSO access type unconditionally

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/clocksource/mmio.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/clocksource/mmio.c b/drivers/clocksource/mmio.c
index b86162a23386..3e5ed0d9974b 100644
--- a/drivers/clocksource/mmio.c
+++ b/drivers/clocksource/mmio.c
@@ -463,9 +463,7 @@ int __init clocksource_user_mmio_init(struct clocksource_user_mmio *ucs,
 		goto fail;
 	}
 
-#ifdef CONFIG_GENERIC_CLOCKSOURCE_VDSO
 	ucs->mmio.clksrc.vdso_type = CLOCKSOURCE_VDSO_MMIO + ucs->id;
-#endif
 
 	if (user_mmio_class) {
 		err = ucs_create_cdev(user_mmio_class, ucs);
-- 
2.16.4

