From 287e86bf642e5b39e95ae8af6a1b2d53820ff406 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 12 Aug 2019 12:12:38 +0200
Subject: [PATCH] dovetail: altsched: prevent the compiler from reordering past
 switch_to()

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/sched/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index c5fd6bcee496..eaba34c0ae0e 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -8109,6 +8109,7 @@ bool dovetail_context_switch(struct dovetail_altsched_context *out,
 	}
 
 	switch_to(prev, next, last);
+	barrier();
 
 	if (check_hard_irqs_disabled())
 		hard_irqs_disabled();
-- 
2.16.4

