From 3793855a035cf875eeb7ff3eaabbf8b82ef52dfa Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 11 Jul 2019 21:32:28 +0200
Subject: [PATCH] dovetail: postpone check for mayday signal

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/irq_pipeline.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/linux/irq_pipeline.h b/include/linux/irq_pipeline.h
index 8ddda27db0e6..f56568cf7bc3 100644
--- a/include/linux/irq_pipeline.h
+++ b/include/linux/irq_pipeline.h
@@ -56,6 +56,9 @@ static __always_inline void enter_irq_pipeline(struct pt_regs *regs)
 
 void exit_oob_irq(void);
 
+void dovetail_call_mayday(struct thread_info *ti,
+			  struct pt_regs *regs);
+
 static __always_inline bool leave_irq_pipeline(struct pt_regs *regs)
 {
 	exit_oob_irq();
@@ -78,6 +81,15 @@ static __always_inline bool leave_irq_pipeline(struct pt_regs *regs)
 	 */
 	synchronize_pipeline_on_irq();
 
+#ifdef CONFIG_DOVETAIL
+	/*
+	 * Sending MAYDAY is in essence a rare case, so prefer test
+	 * then maybe clear over test_and_clear.
+	 */
+	if (user_mode(regs) && test_thread_flag(TIF_MAYDAY))
+		dovetail_call_mayday(current_thread_info(), regs);
+#endif
+
 	return running_inband() && !irqs_disabled();
 }
 
-- 
2.16.4

