From 21ecb4da69d94bdb177b0bba4a07811c4618af8a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 12 Aug 2019 12:12:38 +0200
Subject: [PATCH] dovetail: altsched: prevent the compiler from reordering past
 switch_to()

---
 kernel/sched/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index a7e3a2cff1c5..856f33dd5f88 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -8099,6 +8099,7 @@ bool dovetail_context_switch(struct dovetail_altsched_context *out,
 	}
 
 	switch_to(prev, next, last);
+	barrier();
 
 	if (check_hard_irqs_disabled())
 		hard_irqs_disabled();
-- 
2.16.4

