From 6e9fc0584ceeb5b6e1f65ec02301f6e5b5d5ebee Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 5 Jul 2019 11:54:20 +0200
Subject: [PATCH] genirq: irq_pipeline: fix 2-level mapping helper

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/irq/pipeline.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/kernel/irq/pipeline.c b/kernel/irq/pipeline.c
index 5d2c18c8a22d..6785cfd83ac3 100644
--- a/kernel/irq/pipeline.c
+++ b/kernel/irq/pipeline.c
@@ -624,15 +624,20 @@ static inline int pull_next_irq(struct irq_stage_data *p)
 
 #else /* __IRQ_STAGE_MAP_LEVELS == 2 */
 
-static void clear_pending_irq(struct irq_stage *stage, unsigned int irq)
+static void __clear_pending_irq(struct irq_stage_data *p, unsigned int irq)
 {
-	struct irq_stage_data *p = this_staged(stage);
 	int l0b = irq / BITS_PER_LONG;
 
 	__clear_bit(irq, p->log.map->lomap);
 	__clear_bit(l0b, &p->log.himap);
 }
 
+static void clear_pending_irq(struct irq_stage *stage, unsigned int irq)
+{
+	struct irq_stage_data *p = this_staged(stage);
+	__clear_pending_irq(p, irq);
+}
+
 /* Must be called hw IRQs off. */
 void irq_post_stage(struct irq_stage *stage, unsigned int irq)
 {
@@ -828,12 +833,6 @@ static void handle_unexpected_irq(struct irq_desc *desc, irqreturn_t ret)
 
 	desc->last_unhandled = jiffies;
 
-	/*
-	 * If more than 1000 unhandled events were received
-	 * consecutively, we have to stop this IRQ from poking us at
-	 * the oob of the pipeline by disabling out-of-band mode for
-	 * the interrupt.
-	 */
 	if (unlikely(desc->irqs_unhandled > 1000)) {
 		printk(KERN_ERR "out-of-band irq %d: stuck or unexpected\n", irq);
 		irq_settings_clr_oob(desc);
-- 
2.16.4

