From 56401df0df082c680268df0065e0bbf0cb582fa3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 21 Jul 2016 11:43:21 +0200
Subject: [PATCH] genirq: irq_pipeline: enable pipelining for generic IRQ chip

---
 include/linux/irq.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/irq.h b/include/linux/irq.h
index e27326ebe8e..056f0d35cde 100644
--- a/include/linux/irq.h
+++ b/include/linux/irq.h
@@ -990,7 +990,7 @@ struct irq_chip_type {
  * different flow mechanisms (level/edge) for it.
  */
 struct irq_chip_generic {
-	raw_spinlock_t		lock;
+	hard_spinlock_t		lock;
 	void __iomem		*reg_base;
 	u32			(*reg_readl)(void __iomem *addr);
 	void			(*reg_writel)(u32 val, void __iomem *addr);
-- 
2.16.4

