From 14c387bd7ecd2d9e86fa142805c2feb35a5112c6 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 26 Apr 2019 17:55:32 +0200
Subject: [PATCH] lockdep: irq_pipeline: reword configuration warning

---
 lib/Kconfig.debug | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 20c322be3bd8..15c202c33e8d 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1225,7 +1225,7 @@ config DEBUG_LOCK_ALLOC
 
 if IRQ_PIPELINE && PROVE_LOCKING
 comment "WARNING! PROVE_LOCKING induces **massive** latency"
-comment "overhead for the co-kernel running on the head"
+comment "overhead for the code running on the out-of-band"
 comment "interrupt stage."
 endif
 
-- 
2.16.4

