From 33a5ee24d025bd0b92f61061e0f4ab28959f9e11 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 23 Oct 2019 20:22:14 +0200
Subject: [PATCH] locking: irq_pipeline: fix mutable spin trylock

---
 include/linux/spinlock_pipeline.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/spinlock_pipeline.h b/include/linux/spinlock_pipeline.h
index a22b98724a52..75a915531255 100644
--- a/include/linux/spinlock_pipeline.h
+++ b/include/linux/spinlock_pipeline.h
@@ -270,7 +270,7 @@ static inline int mutable_spin_trylock(struct raw_spinlock *rlock)
 {
 	if (in_pipeline()) {
 		if (do_raw_spin_trylock(rlock)) {
-			hard_lock_acquire(rlock, 1, _THIS_IP_);
+			hard_trylock_acquire(rlock, 1, _THIS_IP_);
 			return 1;
 		}
 		return 0;
@@ -287,7 +287,7 @@ int __mutable_spin_trylock_irqsave(struct raw_spinlock *rlock,
 		int __ret = 1;						\
 		if (in_pipeline()) {					\
 			if (do_raw_spin_trylock(__rlock)) {		\
-				hard_lock_acquire(__rlock, 1, _THIS_IP_); \
+				hard_trylock_acquire(__rlock, 1, _THIS_IP_); \
 				(__flags) = hard_local_save_flags();	\
 			} else						\
 				__ret = 0;				\
-- 
2.16.4

