From 50826279dcec860f1fb2278072a130871f42f2b0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 27 May 2019 17:00:29 +0200
Subject: [PATCH] locking: irq_pipeline: fix allnoconfig build issue

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 lib/Kconfig.debug | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index da2b1afc2535..4e5e2caee654 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1126,7 +1126,6 @@ config LOCK_STAT
 	select DEBUG_MUTEXES
 	select DEBUG_RT_MUTEXES if RT_MUTEXES
 	select DEBUG_LOCK_ALLOC
-	select DEBUG_HARD_LOCKS
 	default n
 	help
 	 This feature enables tracking lock contention points
-- 
2.16.4

