From bc4256b32d1253a9874fcd2b685cc516b1aa0958 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 27 May 2019 17:00:29 +0200
Subject: [PATCH] locking: irq_pipeline: fix allnoconfig build issue

---
 lib/Kconfig.debug | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index b395882c0c4f..b3c9452bdbfc 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1171,7 +1171,6 @@ config LOCK_STAT
 	select DEBUG_MUTEXES
 	select DEBUG_RT_MUTEXES if RT_MUTEXES
 	select DEBUG_LOCK_ALLOC
-	select DEBUG_HARD_LOCKS
 	default n
 	help
 	 This feature enables tracking lock contention points
-- 
2.16.4

