From 4aaa9980af5722cd40ce222bb3eded9b0dcab64b Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 27 May 2019 17:00:29 +0200
Subject: [PATCH] locking: irq_pipeline: fix allnoconfig build issue

---
 lib/Kconfig.debug | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 4098b553d33a..79e57977a001 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1154,7 +1154,6 @@ config LOCK_STAT
 	select DEBUG_MUTEXES
 	select DEBUG_RT_MUTEXES if RT_MUTEXES
 	select DEBUG_LOCK_ALLOC
-	select DEBUG_HARD_LOCKS
 	default n
 	help
 	 This feature enables tracking lock contention points
-- 
2.16.4

