From 1a71b38045c4a635d2ea8136687411eee35f6e4a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 26 Apr 2019 17:55:32 +0200
Subject: [PATCH] lockdep: irq_pipeline: reword configuration warning

---
 lib/Kconfig.debug | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 30ce5615895a..d54daf8ce149 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1253,7 +1253,7 @@ config DEBUG_LOCK_ALLOC
 
 if IRQ_PIPELINE && PROVE_LOCKING
 comment "WARNING! PROVE_LOCKING induces **massive** latency"
-comment "overhead for the co-kernel running on the head"
+comment "overhead for the code running on the out-of-band"
 comment "interrupt stage."
 endif
 
-- 
2.16.4

