From fb4181f0f696cf3b4be5e9b4e4a66f3196e5e7e0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 22 Apr 2017 20:52:59 +0200
Subject: [PATCH] notifier: dovetail: enable atomic notifier from oob stage

---
 kernel/notifier.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/notifier.c b/kernel/notifier.c
index 6196af8a822..9c4191de190 100644
--- a/kernel/notifier.c
+++ b/kernel/notifier.c
@@ -190,6 +190,9 @@ NOKPROBE_SYMBOL(__atomic_notifier_call_chain);
 int atomic_notifier_call_chain(struct atomic_notifier_head *nh,
 			       unsigned long val, void *v)
 {
+	if (!running_inband())
+		return notifier_call_chain(&nh->head, val, v, -1, NULL);
+
 	return __atomic_notifier_call_chain(nh, val, v, -1, NULL);
 }
 EXPORT_SYMBOL_GPL(atomic_notifier_call_chain);
-- 
2.16.4

