From 14d3ff8674e73b818b7ddfe3f46051df56f0397f Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 8 May 2017 15:47:44 +0200
Subject: [PATCH] preempt: irq_pipeline: consider hard irq state in preemptible
 check

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/preempt.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/preempt.h b/include/linux/preempt.h
index fcbe263f2496..c688ac6610e4 100644
--- a/include/linux/preempt.h
+++ b/include/linux/preempt.h
@@ -195,7 +195,8 @@ do { \
 
 #define preempt_enable_no_resched() sched_preempt_enable_no_resched()
 
-#define preemptible()	(preempt_count() == 0 && !irqs_disabled())
+#define preemptible()	(preempt_count() == 0 && \
+			 !hard_irqs_disabled() && !irqs_disabled())
 
 #ifdef CONFIG_PREEMPTION
 #define preempt_enable() \
-- 
2.16.4

