From c8b192ddc1613bb80c6ef24f3c888853d39cb4ff Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 24 Apr 2019 17:22:53 +0200
Subject: [PATCH] rcu: irq_pipeline: remove cyclic header dependency

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/rcupdate.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/linux/rcupdate.h b/include/linux/rcupdate.h
index 5652a61632da..c9a5838e7733 100644
--- a/include/linux/rcupdate.h
+++ b/include/linux/rcupdate.h
@@ -24,7 +24,6 @@
 #include <linux/compiler.h>
 #include <linux/atomic.h>
 #include <linux/irqflags.h>
-#include <linux/irqstage.h>
 #include <linux/preempt.h>
 #include <linux/bottom_half.h>
 #include <linux/lockdep.h>
-- 
2.16.4

