From 0da8f7e4b347cd13751220dab8bc36e6273c6ab0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 24 May 2019 19:48:55 +0200
Subject: [PATCH] sched: irq_pipeline: fix naming conflict w/ iSCSI

---
 kernel/sched/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 165dfca19229..15c7ae728005 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -3709,7 +3709,7 @@ static inline void preempt_sync_inband_irqs(void)
 		preempt_disable();
 		trace_hardirqs_on();
 		clear_stage_bit(STAGE_STALL_BIT, p);
-		sync_current_stage();
+		sync_current_irq_stage();
 		preempt_enable_no_resched_notrace();
 	} else
 		clear_stage_bit(STAGE_STALL_BIT, p);
-- 
2.16.4

