From 6cc4eac57a393713928a1f7759ee38cffdfe71ae Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 14 Jul 2019 16:59:50 +0200
Subject: [PATCH] sched: extend interface to synchronous flags

---
 include/linux/thread_info.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/linux/thread_info.h b/include/linux/thread_info.h
index a92b4e58eba..fca31cf2339 100644
--- a/include/linux/thread_info.h
+++ b/include/linux/thread_info.h
@@ -189,6 +189,19 @@ void clear_ti_local_flags(struct thread_info *ti, unsigned int mask)
 	ti_local_flags(ti) &= ~mask;
 }
 
+static inline
+int test_and_clear_ti_local_flags(struct thread_info *ti, unsigned int mask)
+{
+	int old = ti_local_flags(ti) & mask;
+	ti_local_flags(ti) &= ~mask;
+	return old != 0;
+}
+
+static inline int test_and_clear_thread_local_flags(unsigned int mask)
+{
+	return test_and_clear_ti_local_flags(current_thread_info(), mask);
+}
+
 static inline void clear_thread_local_flags(unsigned int mask)
 {
 	clear_ti_local_flags(current_thread_info(), mask);
-- 
2.16.4

