From 794a551905902413cc85c381f0420816470344b4 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 24 Jul 2016 15:19:00 +0200
Subject: [PATCH] smp: irq_pipeline: add preemption-safe hard_{get, put}_cpu()
 ops

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/smp.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/include/linux/smp.h b/include/linux/smp.h
index 6fc856c9eda5..fb5ce68614d6 100644
--- a/include/linux/smp.h
+++ b/include/linux/smp.h
@@ -222,6 +222,21 @@ static inline int get_boot_cpu_id(void)
 #define get_cpu()		({ preempt_disable(); __smp_processor_id(); })
 #define put_cpu()		preempt_enable()
 
+#ifdef CONFIG_IRQ_PIPELINE
+#define hard_get_cpu(flags)	({			\
+		(flags) = hard_preempt_disable();	\
+		raw_smp_processor_id();			\
+	})
+#define hard_put_cpu(flags)	hard_preempt_enable(flags)
+#else
+#define hard_get_cpu(flags)	({ (void)(flags); get_cpu(); })
+#define hard_put_cpu(flags)	\
+	do {			\
+		(void)(flags);	\
+		put_cpu();	\
+	} while (0)
+#endif
+
 /*
  * Callback to arch code if there's nosmp or maxcpus=0 on the
  * boot command line:
-- 
2.16.4

