From e26c124931224ea8e0612e9f2d8efe5e9cec5239 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 7 Oct 2019 11:51:49 +0200
Subject: [PATCH] tty: imx: silence warning with RAW_PRINTK disabled

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/tty/serial/imx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index c08ea42128fa..4eda24ce8813 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1971,11 +1971,13 @@ imx_uart_console_write(struct console *co, const char *s, unsigned int count)
 	return __imx_uart_console_write(co, s, count, 1);
 }
 
+#ifdef CONFIG_RAW_PRINTK
 static void
 imx_uart_console_write_raw(struct console *co, const char *s, unsigned int count)
 {
 	return __imx_uart_console_write(co, s, count, 0);
 }
+#endif
 
 /*
  * If the port was already initialised (eg, by a boot loader),
-- 
2.16.4

