From 204d7f75e2cbde9ac34fb1082c4c0862344aafb3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 22 Aug 2019 10:36:23 +0200
Subject: [PATCH] tty: imx: add raw console channel

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/tty/serial/imx.c | 58 ++++++++++++++++++++++++++++++++++--------------
 1 file changed, 41 insertions(+), 17 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index a9e20e6c63ad..c08ea42128fa 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1903,30 +1903,33 @@ static void imx_uart_console_putchar(struct uart_port *port, int ch)
  * Interrupts are disabled on entering
  */
 static void
-imx_uart_console_write(struct console *co, const char *s, unsigned int count)
+__imx_uart_console_write(struct console *co, const char *s, unsigned int count, int locked)
 {
 	struct imx_port *sport = imx_uart_ports[co->index];
 	struct imx_port_ucrs old_ucr;
 	unsigned int ucr1;
 	unsigned long flags = 0;
-	int locked = 1;
 	int retval;
 
-	retval = clk_enable(sport->clk_per);
-	if (retval)
-		return;
-	retval = clk_enable(sport->clk_ipg);
-	if (retval) {
-		clk_disable(sport->clk_per);
-		return;
+	if (!IS_ENABLED(CONFIG_RAW_PRINTK)) {
+		retval = clk_enable(sport->clk_per);
+		if (retval)
+			return;
+		retval = clk_enable(sport->clk_ipg);
+		if (retval) {
+			clk_disable(sport->clk_per);
+			return;
+		}
 	}
 
-	if (sport->port.sysrq)
-		locked = 0;
-	else if (oops_in_progress)
-		locked = spin_trylock_irqsave(&sport->port.lock, flags);
-	else
-		spin_lock_irqsave(&sport->port.lock, flags);
+	if (locked) {
+		if (sport->port.sysrq)
+			locked = 0;
+		else if (oops_in_progress)
+			locked = spin_trylock_irqsave(&sport->port.lock, flags);
+		else
+			spin_lock_irqsave(&sport->port.lock, flags);
+	}
 
 	/*
 	 *	First, save UCR1/2/3 and then disable interrupts
@@ -1956,8 +1959,22 @@ imx_uart_console_write(struct console *co, const char *s, unsigned int count)
 	if (locked)
 		spin_unlock_irqrestore(&sport->port.lock, flags);
 
-	clk_disable(sport->clk_ipg);
-	clk_disable(sport->clk_per);
+	if (!IS_ENABLED(CONFIG_RAW_PRINTK)) {
+		clk_disable(sport->clk_ipg);
+		clk_disable(sport->clk_per);
+	}
+}
+
+static void
+imx_uart_console_write(struct console *co, const char *s, unsigned int count)
+{
+	return __imx_uart_console_write(co, s, count, 1);
+}
+
+static void
+imx_uart_console_write_raw(struct console *co, const char *s, unsigned int count)
+{
+	return __imx_uart_console_write(co, s, count, 0);
 }
 
 /*
@@ -2067,6 +2084,10 @@ imx_uart_console_setup(struct console *co, char *options)
 	retval = clk_prepare(sport->clk_per);
 	if (retval)
 		clk_unprepare(sport->clk_ipg);
+	else if (IS_ENABLED(CONFIG_RAW_PRINTK)) {
+		clk_enable(sport->clk_ipg);
+		clk_enable(sport->clk_per);
+	}
 
 error_console:
 	return retval;
@@ -2076,6 +2097,9 @@ static struct uart_driver imx_uart_uart_driver;
 static struct console imx_uart_console = {
 	.name		= DEV_NAME,
 	.write		= imx_uart_console_write,
+#ifdef CONFIG_RAW_PRINTK
+	.write_raw	= imx_uart_console_write_raw,
+#endif
 	.device		= uart_console_device,
 	.setup		= imx_uart_console_setup,
 	.flags		= CON_PRINTBUFFER,
-- 
2.16.4

