From 37953fc92b181d302b1f99d59db292f73242fbcc Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 24 Apr 2019 12:42:47 +0200
Subject: [PATCH] x86/thread: export offset of synchronous status

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/x86/kernel/asm-offsets.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/kernel/asm-offsets.c b/arch/x86/kernel/asm-offsets.c
index 5c7ee3df4d0b..eb77b8c2b6d3 100644
--- a/arch/x86/kernel/asm-offsets.c
+++ b/arch/x86/kernel/asm-offsets.c
@@ -38,6 +38,7 @@ static void __used common(void)
 #endif
 
 	BLANK();
+	OFFSET(TASK_TI_status, task_struct, thread_info.status);
 	OFFSET(TASK_addr_limit, task_struct, thread.addr_limit);
 
 	BLANK();
-- 
2.16.4

