From d7285063aef7c183656ec13df5930d8064eb4b1c Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 30 Jul 2019 15:44:39 +0200
Subject: [PATCH] x86: irq_pipeline: skip irq_enter/exit from oob stage

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/x86/include/asm/apic.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/x86/include/asm/apic.h b/arch/x86/include/asm/apic.h
index 9094b787b525..82a200da3733 100644
--- a/arch/x86/include/asm/apic.h
+++ b/arch/x86/include/asm/apic.h
@@ -515,12 +515,12 @@ static inline bool apic_id_is_primary_thread(unsigned int id) { return false; }
 static inline void apic_smt_update(void) { }
 #endif
 
-extern void irq_enter(void);
-extern void irq_exit(void);
+extern void irq_enter_if_inband(void);
+extern void irq_exit_if_inband(void);
 
 static inline void entering_irq(void)
 {
-	irq_enter();
+	irq_enter_if_inband();
 	kvm_set_cpu_l1tf_flush_l1d();
 }
 
@@ -532,20 +532,20 @@ static inline void entering_ack_irq(void)
 
 static inline void ipi_entering_ack_irq(void)
 {
-	irq_enter();
+	irq_enter_if_inband();
 	ack_APIC_irq();
 	kvm_set_cpu_l1tf_flush_l1d();
 }
 
 static inline void exiting_irq(void)
 {
-	irq_exit();
+	irq_exit_if_inband();
 }
 
 static inline void exiting_ack_irq(void)
 {
 	ack_APIC_irq();
-	irq_exit();
+	irq_exit_if_inband();
 }
 
 extern void ioapic_zap_locks(void);
-- 
2.16.4

