From 6d087ef9c373fa9cd2ca8a4d43c66465145a8377 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 28 Apr 2019 16:35:02 +0200
Subject: [PATCH] x86/vdso: dovetail: serialize gtod update vs read

---
 arch/x86/include/asm/vgtod.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/include/asm/vgtod.h b/arch/x86/include/asm/vgtod.h
index 913a133f8e6f..d1bafd04cdcd 100644
--- a/arch/x86/include/asm/vgtod.h
+++ b/arch/x86/include/asm/vgtod.h
@@ -80,6 +80,7 @@ static inline int gtod_read_retry(const struct vsyscall_gtod_data *s,
 
 static inline void gtod_write_begin(struct vsyscall_gtod_data *s)
 {
+	hard_cond_local_irq_disable();
 	++s->seq;
 	smp_wmb();
 }
@@ -88,6 +89,7 @@ static inline void gtod_write_end(struct vsyscall_gtod_data *s)
 {
 	smp_wmb();
 	++s->seq;
+	hard_cond_local_irq_enable();
 }
 
 #endif /* _ASM_X86_VGTOD_H */
-- 
2.16.4

