From cf0e877320e2761cf68c1d718b076a65313c9563 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 27 May 2019 11:40:38 +0200
Subject: [PATCH] x86: irq_pipeline: fix build issues in APIC event trampoline

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/x86/kernel/irq_pipeline.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/irq_pipeline.c b/arch/x86/kernel/irq_pipeline.c
index f71eae8eee02..391e76bc2e33 100644
--- a/arch/x86/kernel/irq_pipeline.c
+++ b/arch/x86/kernel/irq_pipeline.c
@@ -79,6 +79,8 @@ void irq_pipeline_send_remote(unsigned int ipi,
 }
 EXPORT_SYMBOL_GPL(irq_pipeline_send_remote);
 
+void uv_bau_message_interrupt(struct pt_regs *regs);
+
 static void do_apic_irq(unsigned int irq, struct pt_regs *regs)
 {
 	int vector = apicm_irq_vector(irq);
@@ -138,7 +140,7 @@ static void do_apic_irq(unsigned int irq, struct pt_regs *regs)
 	case HYPERV_REENLIGHTENMENT_VECTOR:
 		hyperv_reenlightenment_intr(regs);
 		break;
-	case HYPERV_STIMER0_VECTOR
+	case HYPERV_STIMER0_VECTOR:
 		hv_stimer0_vector_handler(regs);
 		break;
 #endif
-- 
2.16.4

