From d877117579cb43d2e1f13feea78efe225bc7abb4 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 6 Feb 2020 15:23:51 +0100
Subject: [PATCH] x86: dovetail: update call to pipeline_syscall()

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/x86/entry/common.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/arch/x86/entry/common.c b/arch/x86/entry/common.c
index d077be9f2d10..b3fa28781ec7 100644
--- a/arch/x86/entry/common.c
+++ b/arch/x86/entry/common.c
@@ -312,7 +312,7 @@ __visible void do_syscall_64(unsigned long nr, struct pt_regs *regs)
 	enable_local_irqs();
 	ti = current_thread_info();
 
-	ret = pipeline_syscall(ti, nr, regs);
+	ret = pipeline_syscall(nr, regs);
 	if (ret > 0) {
 		disable_local_irqs();
 		return;
@@ -342,8 +342,8 @@ __visible void do_syscall_64(unsigned long nr, struct pt_regs *regs)
 #if defined(CONFIG_X86_32) || defined(CONFIG_IA32_EMULATION)
 
 #if defined(CONFIG_DOVETAIL) && defined(CONFIG_X86_64)
-static inline int pipeline_syscall32(struct thread_info *ti,
-				unsigned long nr, struct pt_regs *regs)
+static inline
+int pipeline_syscall32(unsigned long nr, struct pt_regs *regs)
 {
 	struct pt_regs regs64 = *regs;
 	int ret;
@@ -353,14 +353,14 @@ static inline int pipeline_syscall32(struct thread_info *ti,
 	regs64.r10 = (unsigned int)regs->si;
 	regs64.r8 = (unsigned int)regs->di;
 	regs64.r9 = (unsigned int)regs->bp;
-	ret = pipeline_syscall(ti, nr, &regs64);
+	ret = pipeline_syscall(nr, &regs64);
 	regs->ax = (unsigned int)regs64.ax;
 
 	return ret;
 }
 #else
-static inline int pipeline_syscall32(struct thread_info *ti,
-				   unsigned long nr, struct pt_regs *regs)
+static inline
+int pipeline_syscall32(unsigned long nr, struct pt_regs *regs)
 {
 	return 0;
 }
@@ -382,7 +382,7 @@ static __always_inline void do_syscall_32_irqs_on(struct pt_regs *regs)
 	ti->status |= TS_COMPAT;
 #endif
 
-	ret = pipeline_syscall32(ti, nr, regs);
+	ret = pipeline_syscall32(nr, regs);
 	if (ret > 0) {
 		disable_local_irqs();
 		return;
-- 
2.16.4

