From a37dc5ad846403141daf958fdc159fb10dc4e417 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 24 May 2019 19:53:29 +0200
Subject: [PATCH] x86: irq_pipeline: fix naming conflict w/ iSCSI

---
 arch/x86/entry/entry_64.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/entry/entry_64.S b/arch/x86/entry/entry_64.S
index 5b65448f0680..fcee162ab172 100644
--- a/arch/x86/entry/entry_64.S
+++ b/arch/x86/entry/entry_64.S
@@ -568,7 +568,7 @@ ENTRY(interrupt_entry)
 	/*
 	 * We entered an interrupt context - irqs are off unless
 	 * pipelining is enabled, in which case we defer tracing until
-	 * sync_current_stage() is called from the inband stage.
+	 * sync_current_irq_stage() is called from the inband stage.
 	 */
 #ifndef CONFIG_IRQ_PIPELINE
 	TRACE_IRQS_OFF
-- 
2.16.4

