From 48678b03384641390e3c604c1147d0ce38127bfe Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 31 Dec 2019 12:26:20 +0100
Subject: [PATCH] arm64/build: evl: drop pointless include directives

---
 arch/arm64/Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 8ad5937808b3..dca1a97751ab 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -84,10 +84,6 @@ endif
 
 CHECKFLAGS	+= -D__aarch64__
 
-ifeq ($(CONFIG_EVL),y)
-KBUILD_CFLAGS += -Iarch/$(SRCARCH)/evl/include -Iinclude/evl
-endif
-
 ifeq ($(CONFIG_ARM64_MODULE_PLTS),y)
 KBUILD_LDS_MODULE	+= $(srctree)/arch/arm64/kernel/module.lds
 endif
-- 
2.16.4

