From 1584d47323c960aad179fe251b8bf6475e4563fe Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 5 Oct 2019 18:05:06 +0200
Subject: [PATCH] evl/thread: pull inband sched ABI details from uapi

---
 kernel/evl/thread.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/evl/thread.c b/kernel/evl/thread.c
index e528065e3fdb..d8c875686855 100644
--- a/kernel/evl/thread.c
+++ b/kernel/evl/thread.c
@@ -22,6 +22,7 @@
 #include <linux/err.h>
 #include <linux/ptrace.h>
 #include <linux/math64.h>
+#include <uapi/linux/sched/types.h>
 #include <evl/sched.h>
 #include <evl/timer.h>
 #include <evl/wait.h>
-- 
2.16.4

